usaspending_api/accounts/views/federal_accounts_v2.py

```diff
@@ -49,3 +63,295 @@ def get(self, request, pk, format=None):
             for maj in major_classes
         ]
         return Response({'results': result})
+
+
+class FiscalYearSnapshotFederalAccountsViewSet(APIView):
+    @cache_response()
+    def get(self, request, pk, format=None):
+
+        queryset = AppropriationAccountBalances.final_objects.filter(treasury_account_identifier__federal_account_id=int(
+            pk)).filter(submission__reporting_fiscal_year=fy(datetime.today()))
+        queryset = queryset.aggregate(
+            outlay=Sum('gross_outlay_amount_by_tas_cpe'),
+            budget_authority=Sum('budget_authority_available_amount_total_cpe'),
+            obligated=Sum('obligations_incurred_total_by_tas_cpe'),
+            unobligated=Sum('unobligated_balance_cpe'),
+            balance_brought_forward=Sum(
+                    F('budget_authority_unobligated_balance_brought_forward_fyb')
+                    +
+                    F('adjustments_to_unobligated_balance_brought_forward_cpe')
+                    ),
+            other_budgetary_resources=Sum('other_budgetary_resources_amount_cpe'),
+            appropriations=Sum('budget_authority_appropriated_amount_cpe')
+            )
+        if queryset['outlay'] is not None:
+            name = FederalAccount.objects.filter(id=int(pk)).values('account_title').first()['account_title']
+            queryset['name'] = name
+            return Response({'results': queryset})
+        else:
+            return Response({})
+
+
+class SpendingOverTimeFederalAccountsViewSet(APIView):
+    @cache_response()
+    def post(self, request, pk, format=None):
+        # create response
+        response = {'results': {}}
+
+        # get federal account id from url
+        json_request = request.data
+        group = json_request.get('group', None)
+        filters = json_request.get('filters', None)
+        if filters:
+            filters = federal_account_filter2(filters, "treasury_account_identifier__program_balances__")
+
+        nested_order = ''
+        group_results = OrderedDict()  # list of time_period objects ie {"fy": "2017", "quarter": "3"} : 1000
+
+        financial_account_queryset = AppropriationAccountBalances.final_objects.filter(treasury_account_identifier__federal_account_id=int(pk))
+        if group == 'fy' or group == 'fiscal_year':
+
+            filtered_fa = financial_account_queryset
```

Comment:
`filtered_fa` and `unfiltered_fa` look like they can be moved outside of the `if` statement and shared by both conditions (the fiscal year would have the quarter value and not use it) but it would save code. Really it looks like a bunch of the code can be reused for both conditions with the major difference being the `group_results` key.